src/main/java/com/faforever/client/vault/SpecialCaseForMapVaultLocationChecker.java

```diff
@@ -0,0 +1,60 @@
+package com.faforever.client.vault;
+
+import com.faforever.client.i18n.I18n;
+import com.faforever.client.notification.Action;
+import com.faforever.client.notification.NotificationService;
+import com.faforever.client.notification.PersistentNotification;
+import com.faforever.client.notification.Severity;
+import com.faforever.client.preferences.ForgedAlliancePrefs;
+import com.faforever.client.preferences.PreferencesService;
+import com.google.common.base.CharMatcher;
+import com.sun.security.auth.module.NTSystem;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+
+import static org.bridj.Platform.isUnix;
+import static org.bridj.Platform.isWindows;
+
+/**
+ * Checks if the user is on linux or has a none ascii username. In this cases we need to use a fallback location for the
+ * vault.
+ */
+@Component
+public class SpecialCaseForMapVaultLocationChecker implements InitializingBean {
```

Comment:
I would not make this an InitializingBean. How do you make sure that the window is loaded already?
Also, some bean should be responsible for calling this check. (Maybe the init method of the MainController ?)